go/stellar/stellarsvc/frontend.go

```diff
@@ -239,16 +243,209 @@ func (s *Server) LinkNewWalletAccountLocal(ctx context.Context, arg stellar1.Lin
 	return accountID, nil
 }
 
+func (s *Server) GetPaymentsLocal(ctx context.Context, arg stellar1.GetPaymentsLocalArg) (payments []stellar1.PaymentOrErrorLocal, err error) {
+	ctx = s.logTag(ctx)
+	defer s.G().CTraceTimed(ctx, "GetPaymentsLocal", func() error { return err })()
+	err = s.assertLoggedIn(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	srvPayments, err := s.remoter.RecentPayments(ctx, arg.AccountID, 0)
+	if err != nil {
+		return nil, err
+	}
+	payments = make([]stellar1.PaymentOrErrorLocal, len(srvPayments))
+	for i, p := range srvPayments {
+		payments[i].Payment, err = s.transformPaymentSummary(ctx, p)
+		if err != nil {
+			s := err.Error()
+			payments[i].Err = &s
+			payments[i].Payment = nil // just to make sure
+		}
+	}
+
+	return payments, nil
+}
+
+func (s *Server) transformPaymentSummary(ctx context.Context, p stellar1.PaymentSummary) (*stellar1.PaymentLocal, error) {
```

Comment:
All this transform stuff would be better suited to live in `stellar/`. So that if some other part outside the RPC handler (like caching / storage) wants to use it that would be easier. I don't feel strongly though.